### ISSUE TITLE
Surface Construction Index Output Variable not found and other DisplayAdvancedReportVariables issues

### ISSUE THREAD

**Issue by XiwangLi** [NREL/EnergyPlus#6155]
Issue overview
--------------
Running the example file "EMSThermochromicWindow.idf" in V8.6.

The output variable "VarName=SURFACE CONSTRUCTION INDEX" is not find. The ".rdd" file does not contain this "SURFACE CONSTRUCTION INDEX". So How can I output this variable?

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)  Windows 7
 - Version of EnergyPlus (if using an intermediate build, include SHA)  8.6
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
@xiwanglienergy This output requires "Output:Diagnostics, DisplayAdvancedReportVariables;" in the idf.  

Several things need to be fixed here.

- [ ] Example file EMSThermochromicWindow.idf needs "Output:Diagnostics, DisplayAdvancedReportVariables;"
- [ ] SURFACE CONSTRUCTION INDEX is not described in the Input Output Reference.
- [ ] The following warning should be modified:
```
  ** Warning ** The following Report Variables were requested but not generated
   **   ~~~   ** because IDF did not contain these elements or misspelled variable name -- check .rdd file
   ************* Key=Perimeter_ZN_1_wall_south_Window_1, VarName=SURFACE CONSTRUCTION INDEX, Frequency=Timestep
 ```
Suggested change:
```
  ** Warning ** The following Report Variables were requested but not generated -- check .rdd file
   **   ~~~   ** Either the IDF did not contain these elements, the variable name is misspelled, 
   **   ~~~   ** or the requested variable is an advanced output which requires Output:Diagnostics, DisplayAdvancedReportVariables;
   ************* Key=Perimeter_ZN_1_wall_south_Window_1, VarName=SURFACE CONSTRUCTION INDEX, Frequency=Timestep
 ```

- [ ] Or consider eliminating DisplayAdvancedReportVariables?  There are approximately 120 variables that fall in this category.  This is the I/O Ref description for that:

> DisplayAdvancedReportVariables – use this to be able to use certain advanced output
variables where the name may be misleading and you need to understand the concepts
or reasons for use. If you put in this field, then you will be able to report on these features.
They are noted in the descriptions of objects or output variables.

- [ ] Searching the code reveals that there are three places where eio output is activated when `DisplayAdvancedReportVariables` is true.  This should be documented in the I/O Ref , or better yet changed to use new Output:* objects.  The following places activate extra eio output:
[Convection geometry](https://github.com/NREL/EnergyPlus/blob/88e72df8e76d49d361e41b3e023d039aa14e7dc3/src/EnergyPlus/ConvectionCoefficients.cc#L5310-L5311)
[Curve manager](https://github.com/NREL/EnergyPlus/blob/88e72df8e76d49d361e41b3e023d039aa14e7dc3/src/EnergyPlus/CurveManager.cc#L3318) sets `EchoTableDataToEio` equal to `DisplayAdvancedReportVariables` which includes echo *all* of the table data - could be a large amount of information.  The use of DisplayAdvancedReportVariables for this *is* mentioned several places in the "Group Performance Tables" section of the I/O Ref.
Surface View Factor Check Values [here and elsewhere in this file](https://github.com/NREL/EnergyPlus/blob/88e72df8e76d49d361e41b3e023d039aa14e7dc3/src/EnergyPlus/HeatBalanceIntRadExchange.cc#L561)

**Comment by lgu1234** [comment]
@mjwitte I assign this issue to me. I would like to fix first 3 items for this issue. The last two items may need consideration and discussion. I would like to make two separate issues for last two items. What do you think?

**Comment by mjwitte** [comment]
@lgu1234 Yes, they should probably be split out to separate issues.

**Comment by lgu1234** [comment]
@mjwitte The output variable of Surface Construction Index reports construction index, which is constant and independent of time. We may not report the output variable with respect to time. Instead, we may report this variable under Detailed Surface Data Report with construction name only and no index in eio, because it is not easy to recognize the index. Please let me know what you think.

**Comment by mjwitte** [comment]
@lgu1234 It can change with time via EMS, so it should stay as an output variable.

**Comment by XiwangLi** [comment]
The construction index may change with respect to time, which is not a
constant when we do thermochromic control. I did output the index through
EMS as a global variable.

**Comment by lgu1234** [comment]
Thanks for your quick reply. OK. Keep it as is.

**Comment by Myoldmopar** [comment]
Closed via #6174
